channels/applicationinsights-channel-js/src/Statsbeat.ts

```diff
@@ -0,0 +1,203 @@
+        dynamicProto(Statsbeat, this, (_self, _base) => {
+            _self.initialize = (config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?:ITelemetryPluginChain, endpoint?: string) => {
+                _networkCounter = new NetworkStatsbeat(endpoint);
+                _sender = new Sender(undefined);
+                let senderConfig = {...config};
+                senderConfig.instrumentationKey = Statsbeat.INSTRUMENTATION_KEY;
+                _sender.initialize(senderConfig, core, extensions, pluginChain, endpoint);
+                _statsbeatMetrics = {};
+                _config = config;
+                _isEnabled = _config.disableStatsbeat !== true;
+                if (_isEnabled) {
+                    _getCustomProperties();
+                    if (!_handle) {
+                        _handle = setInterval(() => {
+                            this.trackShortIntervalStatsbeats();
+                        }, Statsbeat.STATS_COLLECTION_SHORT_INTERVAL);
+                    }
+                }
+            }
+
+            _self.isInitialized = (): boolean => {
+                return !!_isEnabled;
+            }
+
+            _self.countRequest = (endpoint: string, duration: number, success: boolean) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.totalRequestCount++;
+                counter.intervalRequestExecutionTime += duration;
+                if (success === false) {
+                    counter.totalFailedRequestCount++;
+                }
+                else {
+                    counter.totalSuccesfulRequestCount++;
+                }
+            }
+
+            _self.countException = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.exceptionCount++;
+            }
+            
+            _self.countThrottle = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.throttleCount++;
+            }
+
+            _self.countRetry = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.retryCount++;
+            }
+
+            _self.trackShortIntervalStatsbeats = (): void => {
+                _trackRequestDuration();
+                _trackRequestsCount();
+                _sendStatsbeats();
+            }
+
+            function _getCustomProperties() {
+                _cikey = _config.instrumentationKey;
```

Comment:
Missing some properties?

Reply:
Yeah... thanks for pointing it out, this is a todo item I forgot to mention. I'll follow up with Nev and Hector:
- what should "language" property be for web? "JavaScript"?
- what should "sdkVersion" property be for web? 
- what about os and runtimeVersion?

Reply:
My 2c
- Language => JavaScript
- sdkVersion => The current sdk version (we set this in the ext.sdk.version)
- os/runTimeVersion => Either empty or Browser (We don't really want to start completely cracking the UserAgent to figure this out), and collector will do that for use anyway.